be/src/olap/base_tablet.cpp

```diff
@@ -548,6 +553,15 @@ Status BaseTablet::lookup_row_key(const Slice& encoded_key, bool with_seq_col,
 
         for (auto id : picked_segments) {
             Status s = segments[id]->lookup_row_key(encoded_key, with_seq_col, with_rowid, &loc);
+            SegmentCache::CacheKey cache_key(rs->rowset_id(), segments[id]->id());
+            auto cache_value = std::make_unique<SegmentCache::CacheValue>();
+            cache_value->segment = segments[id];
+            try {
+                SegmentLoader::instance()->update_segment(cache_key, *cache_value);
```

Comment:
`load_pk_index_and_bf` only would be called once, but `lookup_row_key` might be called for billion times, if you also call `update_segment` for so many times ,the LRU write lock would become bottleneck.